### DISCUSSION TITLE
Postman to Bruno import, what commands do you use ?

### DISCUSSION THREAD

**Discussion by baptpln** [usebruno/bruno#1096] [General]
Currently working on improving the **import collection** experience, especially from Postman.
We are adding automatic imports for scripts and tests sections, instead of incorporating everything inside comments.
Currently we can translate set and get for _pm.variable_ and _pm.environment_
What other commands do you **extensively** use ?
Thanks for your answers.

**Comment by Venya331** [comment]
Hi There) i suppose the most used are
var
eval()
pm.response.json()
tests[\"Status code is 200\"] = responseCode.code === 200
let 
for
if
some fields of json data 
jsonData.<field>.<array>.length
and may be something like that)
pm.test(\"some test name\", () => {\r",
									"            pm.expect(\r",
									"                jsonData.array[i].array2[0].startDate\r",
									"            ).to.have.property('year', somevariable.CRYear);\r",
									"        })

**Reply by baptpln** [reply]
@Venya331 Thanks for your answer, i had in mind commands starting with "pm.", because we can already use pretty much everything from basic javascript commands, the test command and the res object.

**Comment by Venya331** [comment]
oooh) ok got you)
then these ones) 
pm.test()
pm.expect()
pm.response.json()
pm.environment.set()
pm.collectionVariables.set()
pm.collectionVariables.has()
pm.collectionVariables.get()
pm.environment.get()
pm.sendRequest()
pm.globals.get()

**Comment by amaillet2024** [comment]
we also use a lot the iteration so : 
pm.iterationData.get


and this specific fundtion : 
postman.setNextRequest (very important to us to adapt the test case to the iteration data content)

the other function have already been discussed
pm.collectionVariables.clear
pm.test
pm.collectionVariables.set
pm.response.to.have.jsonSchema

**Reply by baptpln** [reply]
Thanks for your reply !

Well bru.setNextRequest() has already been implemented, and I'm working on collection variables it might help a lot of users

**Reply by amaillet2024** [reply]
what a fast answer ! thanks. We are in reflection for migration from postman, We will probably wait a litte to migrate the script with iteration data.

**Comment by lanthoor** [comment]
We use these as well, in addition to the already listed ones.
- `pm.variables.replaceIn()`
- `{{$guid}}` etc. from the https://learning.postman.com/docs/writing-scripts/script-references/variables-list. These are also used in the request body.
- `pm.environment.has()`

**Comment by georgekaz** [comment]
Hi. I tried to import one of our sets of Postman tests and these are the commands it's been unable to handle, including their occurrence count. It would be great if these could be addressed or advice given on how we could replace these manually to work with Bruno.
```
      8 pm.environment.unset
    139 pm.expect
    559 pm.globals.get
    390 pm.globals.set
     91 pm.globals.unset
     44 pm.info.requestName
      6 pm.request.headers.add
     11 pm.request.headers.remove
    103 pm.request.headers.upsert
      1 pm.response.headers.get
      3 pm.response.to.have.header
    124 pm.sendRequest
      9 pm.test
      2 pm.variables.get
      2 pm.variables.replaceIn
  ```

**Comment by jwedel** [comment]
Anyone here succeeded?

Would be great to have a Postman compatibility layer OR maybe some automatic migration or at least a mapping table on how to fix them. We have a lot of test scripts that use those postman commands / variables.

**Comment by kaemelle** [comment]
In postman you don't have the notion of asserts. So a lot of my postman test script should be transformed in asserts :
________________________________________________________________________________________
- 200 response
pm.test("Response status code is 200", function () {
    pm.response.to.have.status(200);
});
=> 
            "assertions": [
              {
                "name": "res.status",
                "value": "eq 200",
                "enabled": true,
              },

- Content-type is present
pm.test("Content-Type is present", function () {
    pm.response.to.have.header("Content-Type");
});
=>
              {
                "name": "res.headers['content-type']",
                "value": "isDefined",
                "enabled": true,
              },

- JSON response format
pm.test("Content-Type header is application/json", () => {
  pm.expect(pm.response.headers.get('Content-Type')).to.include('application/json');
});
=> 
              {
                "name": "res.headers['content-type']",
                "value": "eq application/json",
                "enabled": true,
              },

- Body contains string
pm.test("Body contains string  -", function () {
    pm.expect(pm.response.text()).to.include("-");
});
=>
              {
                "name": "res.body",
                "value": "contains -",
                "enabled": true,
              }
______________________________________________________________
It would be nice if this exemples would be include in the postman auto translator script
